src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2138,10 +2208,11 @@ static List *
 ExtractInsertValuesList(Query *query, Var *partitionColumn)
 {
 	List *insertValuesList = NIL;
-	TargetEntry *targetEntry = get_tle_by_resno(query->targetList,
-												partitionColumn->varattno);
+	RangeTblEntry *valuesRTE = NULL;
 
-	if (targetEntry == NULL)
+	int partitionColumnIndex = GetTargetListEntryIndexByResno(query->targetList,
+															  partitionColumn->varattno);
+	if (partitionColumnIndex == -1)
```

Comment:
I'm assuming this is in response to _The change also clears up some assumptions around the partition column being a Var, which might not be the case_. When is that not the case? This review's description could use a little expansion (in lieu of which I'll probably have to re-investigate this issue and print out query structures myself to see what you're alluding to).